tests/integration/integration_python_test.go

```diff
@@ -1926,6 +1926,44 @@ func TestPythonComponentProviderRun(t *testing.T) {
 	}
 }
 
+// Tests that we can run a Python component provider using bootstrap-less mode.
+//
+//nolint:paralleltest // ProgramTest calls t.Parallel()
+func TestPythonComponentProviderBootstraplessRun(t *testing.T) {
+	testData, err := filepath.Abs(filepath.Join("component_provider", "python", "bootstrap-less"))
+	require.NoError(t, err)
+	providerDir := filepath.Join(testData, "provider")
+	installPythonProviderDependencies(t, providerDir)
+
+	integration.ProgramTest(t, &integration.ProgramTestOptions{
+		Dir: filepath.Join(testData, "yaml"),
+		ExtraRuntimeValidation: func(t *testing.T, stack integration.RuntimeValidationStackInfo) {
+			urn, err := resource.ParseURN(stack.Outputs["urn"].(string))
+			require.NoError(t, err)
+			require.Equal(t, tokens.Type("provider:index:MyComponent"), urn.Type())
+		},
+	})
+}
+
+// Tests that we can run a Python component provider that's a Python package
+//
+//nolint:paralleltest // ProgramTest calls t.Parallel()
```

Comment:
Maybe more importantly:

```suggestion
//nolint:paralleltest // We're installing dependencies in the test data dir
```

Same above.